src/app/cras/edit-cra/edit-cra.component.spec.ts

```diff
@@ -73,70 +81,216 @@ describe('EditCraComponent', () => {
+        testData.mode = 'edit';
+        route.queryParams = of({ data: testEditToken });
+
+        const spySetTitle = spyOn(titleService, 'setTitle'),
+              spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(spySetTitle).toHaveBeenCalledWith('Acrabadabra - Editer un compte rendu d\'activité');
+        expect(spyDeserialize).toHaveBeenCalledWith(testEditToken);
+      });
     });
-  }));
 
-  it('should enable modal and saved text on save', () => {
-    component.cra = testData.cra;
-    component.onSubmitCRA();
-    expect(component.saved).toBe(true);
-    expect(component.showModal).toBe(true);
+    describe('when called in review mode', () => {
+      it('should set mode to review', () => {
+        testData.mode = 'review';
+        route.queryParams = of({ data: testReviewToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(component.mode).toBe(testData.mode);
+        expect(spyDeserialize).toHaveBeenCalledWith(testReviewToken);
+      });
+
+      it('should set Acrabadabra - Consulter un compte rendu d\'activité as page title', () => {
+        testData.mode = 'review';
+        route.queryParams = of({ data: testReviewToken });
+
+        const spySetTitle = spyOn(titleService, 'setTitle'),
+              spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(spySetTitle).toHaveBeenCalledWith('Acrabadabra - Consulter un compte rendu d\'activité');
+        expect(spyDeserialize).toHaveBeenCalledWith(testReviewToken);
+      });
+
+      it('should disable form inputs', () => {
+        testData.mode = 'review';
+        route.queryParams = of({ data: testReviewToken });
+
+        const spyDeserialize = spyOn(serializer, 'deserialize')
+                .and.returnValue(testData);
+
+        fixture.detectChanges();
+        component.ngOnInit();
+
+        expect(component.consultantNameInput.disabled).toBeTruthy();
+        expect(component.consultantEmailInput.disabled).toBeTruthy();
+        expect(component.missionTitleInput.disabled).toBeTruthy();
+        expect(component.missionFinalClientInput.disabled).toBeTruthy();
+        expect(spyDeserialize).toHaveBeenCalledWith(testReviewToken);
+      });
+    });
   });
 
-  it('should save the token', async(() => {
-    component.cra = testData.cra;
-    const spy = spyOn(serializer, 'serialize')
-      .and.returnValue(testEditToken);
-    component.onSubmitCRA();
-    fixture.whenStable().then(() => {
-      expect(component.editToken).toBe(testEditToken);
+  describe('when submiting', () => {
+    describe('and the form is valid', () => {
```

Comment:
What do you mean about this sentence : `with a valid form`.

Always read your entire test to see if the sentence is clear :
```EditCraComponent -  when submiting with a valid form it should...```